File: cpp/gabaritos/gabarito22.cpp
#include <iostream>
#include <stdexcept>

int main() {
    try {
        double a, b;
        std::cout << "Digite o primeiro número: ";
        std::cin >> a;
        std::cout << "Digite o segundo número: ";
        std::cin >> b;
        if (b == 0) {
            throw std::runtime_error("Divisão por zero");
        }
        double resultado = a / b;
        std::cout << "Resultado: " << resultado << std::endl;
    } catch (const std::exception& e) {
        std::cout << "Erro: " << e.what() << std::endl;
    }
    return 0;
}

File: csharp/gabaritos/gabarito22.cs
using System;

class Program
{
    static void Main()
    {
        try
        {
            Console.Write("Digite o primeiro número: ");
            double a = double.Parse(Console.ReadLine());
            Console.Write("Digite o segundo número: ");
            double b = double.Parse(Console.ReadLine());
            double resultado = a / b;
            Console.WriteLine($"Resultado: {resultado}");
        }
        catch (DivideByZeroException)
        {
            Console.WriteLine("Erro: divisão por zero");
        }
        catch (FormatException)
        {
            Console.WriteLine("Erro: entrada inválida");
        }
    }
}